nixos/modules/services/web-apps/gancio.nix

```diff
@@ -0,0 +1,273 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.gancio;
+  isLocalHostName =
+    cfg.hostName == "localhost"
+    || (lib.hasSuffix ".local" cfg.hostName)
+    || (lib.hasSuffix ".localhost" cfg.hostName);
+  json = pkgs.formats.json { };
+in
+{
+  options.services.gancio = {
+    enable = lib.mkEnableOption (
+      lib.mdDoc ''
+        A shared agenda for local communities.
+        The first user to register, using `admin` as email, will be the initial gancio instance administrator.
+      ''
+    );
+
+    package = lib.mkPackageOption pkgs "gancio" { };
+
+    plugins = lib.mkOption {
+      type = with lib.types; listOf package;
+      example = lib.literalExpression "[ pkgs.gancioPlugins.telegram-bridge ]";
+      description = lib.mdDoc ''
+        Paths of gancio plugins to activate (linked under $WorkingDirectory/plugins/).
+      '';
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = lib.mdDoc "The user (and database name) used to run the gancio server";
+      default = "gancio";
+    };
+
+    hostName = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      description = lib.mdDoc "Web hostname to serve gancio on (nginx)";
+    };
+
+    enableACME = lib.mkOption {
+      type = lib.types.bool;
+      default = !isLocalHostName;
+      defaultText = lib.literalMD ''
+        `true`, unless {option}`services.gancio.hostName`
+         is set to localhost or use a local domain.
+      '';
+      description = lib.mdDoc ''
+        Whether an ACME certificate should be used to secure
+        connections to the server.
+      '';
+    };
```

Comment:
I don't understand why we need this option in the service. We can leave it up to people to set services.nginx.virtualHosts."gancio.example.com".enableAcme = true, no?